text/3081-multi-part-examples-in-rustdoc/index.md

```diff
@@ -0,0 +1,230 @@
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Why not an implicit `split-continue`?
+
+We have considered the possibility of the `split-continue` tag either being implicit and therefore possibly omitted or not having such a tag at all.
+Every code block between a `split-start` and a `split-end` would be considered another part of the multi-part example.
+The problem with this design is that without further syntax, it disallows having unrelated code blocks in between multi-part example code blocks.
+For example, if between the `split-start` and `split-end` of a multi-part example, the author wishes to insert a `text` code block—that shows input/output or example data—this would either not be permitted or it will be made possible with some additional logic.
+Both _not permitting_ such nested single-part examples and _additional logic_ for permitting them seems less elegant/desirable than the simpler approach of requiring explicit `split-continue` tags.
+
+## Why error on inconsistent language tags
+
+From the guide section earlier:
+
+> If a language tag is present, it must be the first word in the info-string and must be present in all parts of the multi-part example.
+
+An alternative to this is that specifying the language in the `split-start` block suffices and rustdoc can understand that all following parts of the multi-part example are of the same language.
+There is a problem with that.
+Pure markdown doesn't know and will never know anything about this feature.
+Markdown only [acknowledges][commonmark-fenced] the _typical_ use of the first word to specify a language:
+
+> The first word of the info string is typically used to specify the language of the code sample, and rendered in the class attribute of the code tag. However, this spec does not mandate any particular treatment of the info string.
+
+So any tools that are not rust-specific (or that don't yet implement multi-part examples) would not have the same understanding of this feature as rustdoc will, and therefore will think that only the first code block is whatever language was specified and the rest of the code blocks are not necessarily the same language.
+
+Erroring when the language tag varies between parts of a multi-part example prevents this situation.
+
+At the very least, it would probably prevent some authors from a future moment where they would think to themselves "oh, darn — I wish _I had_ placed the `ruby` tag on _every part_ of the multi-part example".
+
+Since the majority of multi-part examples are expected to be rust code, where the `rust` langauge tag is usually omitted, this error is not expected to be common.
```

Comment:
@jyn514 continuing from https://github.com/rust-lang/rfcs/pull/3081#discussion_r577912128 we were on the fence about this. I, personally, don't like the feature of `rust` being default, for the same compatibility concern. I guess that's where I am coming from in this.

We wouldn't mind if it is deemed more desirable that subsequent code blocks inherit the language of the first one.
If you ask me, I feel it is consistent with a design feature I don't agree with, making things only slightly worse.

Another point is that this inheritance would take a little bit more implementation work, wouldn't it? Not sure it's a relevant point or not. If there is difference in implementation, perhaps we could first error and then re-visit this inheritance later, if anyone really cares.

Reply:
Also continuing https://github.com/rust-lang/rfcs/pull/3081#discussion_r577913807.